internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -248,6 +251,14 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 		}
 	}
 
+	// Adjust the managedFields of the MachineSets to make them compatible with SSA.
+	for idx := range msList {
+		machineSet := msList[idx]
+		if err := ssa.CleanUpManagedFieldsForSSACompatibility(ctx, machineSet, machineDeploymentManagerName, r.Client); err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "failed to adjust the managedFields of the MachineSet %q", machineSet.Name)
+		}
+	}
```

Comment:
Yes and no.

Technically we only need it for a MachineSet that we want to modify with SSA.

If we ever want to get rid of this code again though it would be good to cleanup all managed fields now and then we can drop this code in a few releases.